delira/utils/config.py

```diff
@@ -513,6 +515,135 @@ def create_from_str(cls, data, formatter=yaml.load, decoder_cls=Decoder,
                      **kwargs)
         return config
 
+    def create_argparser(self):
+        '''
+        Creates an argparser for all values in the config
+        Following the pattern: `--training.learning_rate 1234`
+
+        Returns
+        -------
+        argparse.ArgumentParser
+            parser for all variables in the config
+        '''
+        parser = argparse.ArgumentParser(allow_abbrev=False)
+
+        def str2bool(v):
+            if v.lower() in ('yes', 'true', 't', 'y', '1'):
+                return True
+            elif v.lower() in ('no', 'false', 'f', 'n', '0'):
+                return False
+            else:
+                raise argparse.ArgumentTypeError('Boolean value expected.')
+
+        def add_val(dict_like, prefix=''):
+            for key, val in dict_like.items():
+                name = "--{}".format(prefix + key)
+                if val is None:
+                    parser.add_argument(name)
+                else:
+                    if type(val) == bool:
+                        parser.add_argument(name, type=str2bool)
+                    elif isinstance(val, (list, tuple)):
+                        if len(val) > 0 and type(val[0]) != type:
+                            parser.add_argument(name, type=type(val[0]))
+                        else:
+                            parser.add_argument(name)
+                    elif isinstance(val, dict):
+                        add_val(val, prefix=key + '.')
+                    elif isinstance(val, type):
+                        parser.add_argument(name, type=val)
```

Comment:
I doubt that this would work, because you want to get a class-object here, but tell the argparser to expect an instance of the same class...

Reply:
I do not want to pass a class object i want to offer the possibility to just write a `type` into a `Config` and the `argparser` will expect this type then.

Reply:
okay, in that case I'm still not sure, whether this will work, but at least this should be `issubclass` instead of `isinstance` because every object is an instance of `type` but you want to check for class-objects only...

Reply:
Why would i only check for class objects and not types ?

Reply:
Every type is a class object. In python everything is a object (including classes/types), but currently you check for all objects

Reply:
Ok i think i know what you mean i just can't think of an example for better unterstanding.